tests/unit/src/lib/validator.test.js

```diff
@@ -77,36 +83,47 @@ describe("should fail all the bad ninos", () => {
   test.each(bad_prefixes_ninos)(
     "given bad prefix nino of %p, returns null string",
     (ninoArg) => {
-      const result = isValidNino(ninoArg);
-      expect(result).toBeNull();
+      testNinoValidation(ninoArg);
     }
   );
   test.each(bad_unused_prefixes_ninos)(
     "given unused prefix nino of %p, returns null string",
     (ninoArg) => {
-      const result = isValidNino(ninoArg);
-      expect(result).toBeNull();
+      testNinoValidation(ninoArg);
     }
   );
   test.each(bad_suffixes_ninos)(
     "given bad suffix nino of %p, returns null string",
     (ninoArg) => {
-      const result = isValidNino(ninoArg);
-      expect(result).toBeNull();
+      testNinoValidation(ninoArg);
     }
   );
   test.each(bad_length_ninos)(
     "given bad length nino of %p, returns null string",
     (ninoArg) => {
-      const result = isValidNino(ninoArg);
-      expect(result).toBeNull();
+      testNinoValidation(ninoArg);
     }
   );
 });
 
+describe("invoke the invalidCharacters function", () => {
+  it("should return formattedValue when isValidNino does not return the same value", () => {
+    const testValue = "abc def";
+    const formattedValue = testValue.replaceAll(" ", "").toUpperCase();
+    const isValidNinoResult = isValidNino(formattedValue);
+    expect(isValidNinoResult).to.not.equal(formattedValue);
+    const result = invalidCharacters(formattedValue);
+    expect(result).to.equal(formattedValue);
+  });
+});
+
 describe("should return all the good ninos", () => {
   test.each(good_edge_case_ninos)("returns the given nino of %p", (ninoArg) => {
-    const result = isValidNino(ninoArg)[0];
-    expect(result).toEqual(ninoArg.replaceAll(" ", "").toUpperCase());
+    const result = isValidNino(ninoArg);
+    expect(result).to.equal(ninoArg.replaceAll(" ", "").toUpperCase());
+  });
+  test.each(good_edge_case_ninos)("returns the given nino of %p", (ninoArg) => {
```

Comment:
test.each(good_edge_case_ninos)("invalidCharacters  returns the given nino of %p", (ninoArg) => {